doc/releases/changelog-dev.md

```diff
@@ -4,239 +4,477 @@
+      return qml.expval(qml.PauliZ(0))
+
+  with qml.Tracker(dev) as tracker:
+      qml.grad(circuit)(weights)
+  ```
+
+  The tracker can be inspected to extract resource requirements without requiring a 50-qubit circuit
+  run:
+
+  ```pycon
+  >>> tracker.totals
+  {'executions': 451, 'batches': 2, 'batch_len': 451}
+  >>> tracker.history["resources"][0]
+  wires: 50
+  gates: 200
+  depth: 77
+  shots: Shots(total=None)
+  gate_types:
+  {'Rot': 100, 'CNOT': 100}
+  gate_sizes:
+  {1: 100, 2: 100}
+  ```
+
+* Custom operations can now be defined that solely include resource requirements — an explicit
+  decomposition or matrix representation is not needed.
+  [(#4033)](https://github.com/PennyLaneAI/pennylane/pull/4033)
+
+  PennyLane is now able to estimate the total resource requirements of circuits that include one
+  or more of these operations, allowing you to estimate requirements for high-level algorithms 
+  composed of abstract subroutines. 
+
+  These operations can be defined by inheriting from
+  [ResourcesOperation](https://docs.pennylane.ai/en/stable/code/api/pennylane.resource.ResourcesOperation.html)
+  and overriding the `resources()` method to return an appropriate
+  [Resources](https://docs.pennylane.ai/en/stable/code/api/pennylane.resource.Resources.html)
+  object:
+
+  ```python
+  class CustomOp(qml.resource.ResourcesOperation):
+      def resources(self):
+          n = len(self.wires)
+          r = qml.resource.Resources(
+              num_wires=n,
+              num_gates=n ** 2,
+              depth=5,
+          )
+          return r
+  ```
+
+  ```pycon
+  >>> wires = [0, 1, 2]
+  >>> c = CustomOp(wires)
+  >>> c.resources()
+  wires: 3
+  gates: 9
+  depth: 5
+  shots: Shots(total=None)
+  gate_types:
+  {}
+  gate_sizes:
+  {}
+  ```
+  
+  A quantum circuit that contains `CustomOp` can be created and inspected using
+  [qml.specs](https://docs.pennylane.ai/en/stable/code/api/pennylane.specs.html):
+
+  ```python
+  dev = qml.device("default.qubit", wires=wires)
+
+  @qml.qnode(dev)
+  def circ():
+      qml.PauliZ(wires=0)
+      CustomOp(wires)
+      return qml.state()
+  ```
+  
+  ```pycon
+  >>> specs = qml.specs(circ)()
+  >>> specs["resources"].depth
+  6
+  ``` 
+
+<h4>Community contributions from UnitaryHack 🤝</h4>
+
+* [ParametrizedHamiltonian](https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.ParametrizedHamiltonian.html)
+  now has an improved string representation.
+  [(#4176)](https://github.com/PennyLaneAI/pennylane/pull/4176)
+
+  ```pycon
+  >>> def f1(p, t): return p[0] * jnp.sin(p[1] * t)
+  >>> def f2(p, t): return p * t
+  >>> coeffs = [2., f1, f2]
+  >>> observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+  >>> qml.dot(coeffs, observables)
+    (2.0*(PauliX(wires=[0])))
+  + (f1(params_0, t)*(PauliY(wires=[0])))
+  + (f2(params_1, t)*(PauliZ(wires=[0])))
+  ```
+
+<h4>Trace distance is now available in qml.qinfo 💥</h4>
```

Comment:
@isaacdevlugt this is actually one of the UnitaryHack contributions. We could make this line a h5 level, but I think it's better just to remove.